pkgs/development/compilers/llvm/common/compiler-rt/default.nix

```diff
@@ -71,16 +72,21 @@ stdenv.mkDerivation ({
     "-DCMAKE_ASM_COMPILER_TARGET=${stdenv.hostPlatform.config}"
   ] ++ lib.optionals (haveLibc && stdenv.hostPlatform.libc == "glibc") [
     "-DSANITIZER_COMMON_CFLAGS=-I${libxcrypt}/include"
-  ] ++ lib.optionals ((useLLVM || bareMetal || isMusl || isAarch64) && (lib.versions.major release_version == "13")) [
-    "-DCOMPILER_RT_BUILD_LIBFUZZER=OFF"
-  ] ++ lib.optionals (useLLVM || bareMetal || isMusl || isDarwinStatic) [
+  ] ++ lib.optionals (useLLVM && stdenv.cc.libcxx == libcxx) [
+    "-DSANITIZER_CXX_ABI=libcxxabi"
+    "-DSANITIZER_CXX_ABI_LIBNAME=libcxxabi"
+    "-DCOMPILER_RT_USE_BUILTINS_LIBRARY=ON"
+  ] ++ lib.optionals (useLLVM && haveLibc) [
+    "-DCOMPILER_RT_BUILD_SANITIZERS=ON"
+    "-DCOMPILER_RT_BUILD_PROFILE=ON"
+  ] ++ lib.optionals ((useLLVM && !haveLibc) || bareMetal || isMusl || isDarwinStatic) [
     "-DCOMPILER_RT_BUILD_SANITIZERS=OFF"
+    "-DCOMPILER_RT_BUILD_PROFILE=OFF"
```

Comment:
For stuff like this, we're probably best of transitioning to
```
(lib.cmakeBool "COMPILER_RT_BUILD_SANITIZERS" condition)
(lib.cmakeBool "COMPILER_RT_BUILD_SANITIZERS" condition)
```
i.e. instead of conditionally passing flags, we should always pass flags, and conditionally set the values the flags specify.